packages/gguf/scripts/generate-llm.ts

```diff
@@ -0,0 +1,179 @@
+/**
+ * Script for generating llm.ts
+ * The source data is taken from llama.cpp
+ */
+
+import { writeFileSync } from "node:fs";
+
+const SOURCE_CPP_URL = "https://raw.githubusercontent.com/ggerganov/llama.cpp/master/llama.cpp";
```

Comment:
i was basing myself on the `gguf-py/gguf/constants.py` file (i.e. from the Python parser), but I guess the cpp code is at least as good a source of truth?

Reply:
I'm not very familiar with `gguf-py`, but seem like only in cpp code that you can find the list of metadata used by model (in python library, there's only the list of tensors, not list of kv - those can be found in `convert*.py` scripts)

Reply:
yes that makes sense!